usdt-impl/build.rs

```diff
@@ -27,20 +27,46 @@ enum Backend {
     NoOp,
 }
 
+// Extract any -C ARGS ... passed to RUSTFLAGS
+fn extract_encoded_rustc_flags() -> Vec<String> {
+    match env::var_os("CARGO_ENCODED_RUSTFLAGS").as_deref() {
+      Some(rustflags) => {
+          let mut atoms = rustflags.to_str().unwrap_or("").split(' ');
+          let mut flags = vec![];
+          while let Some(atom) = atoms.next() {
+              if atom.starts_with("-C") {
+                  flags.push(atom[2..].to_string());
+              }
+          }
+          flags
+      }
+      _ => vec![]
+    }
+}
+
 fn main() {
     println!("cargo:rerun-if-changed=build.rs");
 
     // `asm` feature was stabilized in 1.59
     let have_stable_asm = version_check::is_min_version("1.59").unwrap_or(false);
     // XXX: `asm_sym` feature is not yet stable
     let have_stable_asm_sym = false;
+    let have_stable_used_with_arg = false;
 
     // Are we being built with a compiler which allows feature flags (nightly)
     let is_nightly = version_check::is_feature_flaggable().unwrap_or(false);
 
     let feat_asm = env::var_os("CARGO_FEATURE_ASM").is_some();
     let feat_strict_asm = env::var_os("CARGO_FEATURE_STRICT_ASM").is_some();
 
+    // Check if upstream have enabled link-dead-code, which in those cases we can
+    // enable standard backend for FreeBSD. We check this by finding the last
+    // -C link-dead-code* flag, and check if it is a negation of link-dead-code
+    let have_link_dead_code = extract_encoded_rustc_flags().iter()
```

Comment:
This is the consumption site I mention above. This logic could be folded into the `extract_encoded_rustc_flags` function to avoid extra work and iteration here. We don't care about _any_ `-C` flags, only the `link-dead-code` flag.